libraries/aws/shadow/src/aws_iot_shadow_subscription.c

```diff
@@ -443,9 +443,9 @@ AwsIotShadowError_t AwsIotShadow_RemovePersistentSubscriptions( IotMqttConnectio
 
         pSubscription = IotLink_Container( _shadowSubscription_t, pSubscriptionLink, link );
 
-        for( i = 0; i < SHADOW_OPERATION_COUNT; i++ )
+        for( i = 0; i < ( uint32_t ) SHADOW_OPERATION_COUNT; i++ )
         {
-            if( ( flags & ( 0x1UL << i ) ) != 0 )
+            if( ( flags & ( 0x1UL << i ) ) != 0U )
```

Comment:
Should `0x1UL` be changed to `0x1U` for consistency?

Reply:
And MISRA 10.6 or 10.7 will probably call out this expression as a 64 bit number operating with a 32 bit number.

Reply:
That might get taken care of later looking at other MISRA rules. In the version of GCC I'm using, Long is 64bits.

Reply:
I do not personally know why this one literal is in hex or long. I also do not know why `i` is 32bits. I do not know what the flags are, if they will ever change etc. I made the smallest change possible to fix the violation and hopefully not create any unintended consequences.